asserts/asserts.go

```diff
@@ -218,11 +219,140 @@ func Decode(serializedAssertion []byte) (Assertion, error) {
 		return nil, fmt.Errorf("parsing assertion headers: %v", err)
 	}
 
-	if len(signature) == 0 {
-		return nil, fmt.Errorf("empty assertion signature")
+	return Assemble(headers, body, content, signature)
+}
+
+// Decoder parses a stream of assertions bundled by ensuring double newlines at the end of each assertion.
+type Decoder struct {
+	rd   io.Reader
+	buf  []byte
+	rest []byte
+}
+
+// NewDecoder returns a Decoder to parse the stream of assertions from the reader.
+func NewDecoder(r io.Reader) *Decoder {
+	return &Decoder{rd: r, buf: make([]byte, 8192)}
+}
+
+func (dec *Decoder) readUntilNLNL() ([]byte, error) {
```

Comment:
Would be much better/simpler to use a bufio.Scanner here:

http://play.golang.org/p/binpjT9Azy